operator/validator/task_executor.go

```diff
@@ -100,17 +101,21 @@ func (c *controller) UpdateFeeRecipient(owner, recipient common.Address) error {
 
 	return nil
 }
-func (c *controller) ExitValidator(pubKey phase0.BLSPubKey, slot phase0.Slot, validatorIndex phase0.ValidatorIndex) error {
-	duty := &spectypes.Duty{
-		Type:           spectypes.BNRoleVoluntaryExit,
+
+func (c *controller) ExitValidator(pubKey phase0.BLSPubKey, blockNumber uint64, validatorIndex phase0.ValidatorIndex) error {
+	logger := c.taskLogger("ExitValidator",
+		fields.PubKey(pubKey[:]),
+		fields.BlockNumber(blockNumber),
+		zap.Uint64("validator_index", uint64(validatorIndex)),
+	)
+
+	c.validatorExitCh <- duties.ExitDescriptor{
```

Comment:
This should probably send in a goroutine like we do with `indicesChanges`, to avoid blocking the `EventHandler` when VoluntaryExitHandler is not currently reading from the channel:
```go
	go func() {
			select {
			case c.indicesChange <- struct{}{}:
			case <-time.After(2 * SlotDuration):
				logger.Error("failed to schedule ExitValidator duty!")
			}
		}()
```
I suggest to include a timeout of 1 or 2 slots with an error log (not debug like we usually do, this is a really bad error that should never happen)